src/osp/Active/SysNewton.cpp

```diff
@@ -37,6 +39,8 @@ using osp::active::ACompNwtBody;
 using osp::active::ACompNwtWorld;
 using osp::active::ACompTransform;
 
+using adera::active::machines::ACompContainerShape;
```

Comment:
Previously I had `ACompContainerShape` defined in ActiveScene, but that doesn't make sense so I moved it to ShipResources.h. The downside is that it requires coupling SysNewton explicitly to the container stuff. Maybe we need a generic "empty shape" primitive enum in the CommonPhysics.h or something?

Reply:
This as a pretty serious issue. There should be a system that converts ACompContainerShape data into collision shapes and mass. Is ACompCollisionShape and ACompMass not able to represent every kind of mass distribution?

Reply:
The root of the problem is that the mass of parts can be distributed over multiple collision volumes, whereas fuel tanks can't. The fuel tank is represented by a single empty, whose volume is used to calculate it capacity. To compute the mass of a fuel tank, you simply check its ACompMass directly. To compute the mass of a part collider, since there may be multiple colliders, you have to compute the total volume of all colliders in the part, divide the part's "massdry" by this volume to get the `partDensity`, and then use the density to calculate the individual mass of each child collider (another name that sounds odd out of context) in order to compute a realistic inertia for the composite part. This means that somewhere, there needs to be a distinction made between the sources of mass.

Alternatively, we could add logic that would distribute the massdry of a part over its collider volumes from the beginning; the tradeoff is that it would no longer be possible to immediately retrieve the massdry of the part from the ACompMass (although it could be stored in the part component or something), but at least that way the masses could all be treated homogeneously and there'd be no need to discriminate between different types of volumes (so we could get rid of the `ACompContainerShape` and have one unified "shape" enum).